src/services/organizationService.ts

```diff
@@ -515,3 +516,88 @@ export async function getOrganizationInstanceFromDelegateEmail(
     return left(error);
   }
 }
+
+export async function getOrganizationFromUserEmail(
+  userEmail: string
+): Promise<Either<Error, Organization>> {
+  try {
+    const userInstance = await User.findOne({
+      include: [
+        {
+          as: "organizations",
+          include: [
+            {
+              as: "users",
+              model: User
+            },
+            {
+              as: "legalRepresentative",
+              model: User
+            }
+          ],
+          model: OrganizationModel,
+          required: true,
+          through: { where: { email: userEmail } }
+        }
+      ],
+      where: { email: userEmail }
+    });
+    if (userInstance === null || !userInstance.organizations) {
+      return left(new Error("Organization not found"));
+    }
+    if (userInstance.organizations.length > 1) {
+      return left(
+        new Error(
+          `DB conflict error: multiple organizations associated to the user ${userEmail}`
+        )
+      );
+    }
+    const errorsOrOrganization: Either<
+      Errors,
+      Organization
+    > = fromOrganizationInstanceToOrganizationObject(
+      userInstance.organizations[0]
+    );
+    return errorsOrOrganization.mapLeft(
+      errors =>
+        new Error(
+          "Invalid organization data. " +
+            errorsToReadableMessages(errors).join(" / ")
+        )
+    );
+  } catch (error) {
+    return left(error);
+  }
+}
+
+export async function getAllRegisteredOrganizations(): Promise<
+  Either<Error, ReadonlyArray<Organization>>
+> {
+  try {
+    const organizationInstances = await OrganizationModel.findAll({
+      include: [
+        {
+          as: "legalRepresentative",
+          model: User
+        }
+      ]
+    });
+
+    const errorsOrOrganizationEithers = organizationInstances.map(
+      fromOrganizationInstanceToOrganizationObject
+    );
+    return right(
+      errorsOrOrganizationEithers.reduce<ReadonlyArray<Organization> | never>(
+        (organizations, errorsOrOrganization) => {
+          if (isLeft(errorsOrOrganization)) {
+            throw new Error("One or more organizations are not a valid object");
```

Comment:
do not throw. return `left( readable errors )` or `right(organization array)`.

you'd better use `array.sequence(either)(errorsOrOrganization)` to stop at first error